src/vs/workbench/services/textfile/common/textFileEditorModel.ts

```diff
@@ -321,12 +321,21 @@ export class TextFileEditorModel extends BaseTextEditorModel implements ITextFil
 			etag = this.lastResolvedDiskStat.etag; // otherwise respect etag to support caching
 		}
 
+		this.taskNotification();
+
+		// Resolve Content
+		return this.textFileService
+			.resolveTextContent(this.resource, { acceptTextOnly: true, etag, encoding: this.preferredEncoding })
+			.then(content => this.handleLoadSuccess(content), error => this.handleLoadError(error));
+	}
+
+	private taskNotification(): void {
 		const storageKey = 'workbench.tasks.ranTaskBefore';
-		const fileName = path.relative(this.storageService['workspaceKey'], this.resource['_formatted']);
+		const fileName = path.relative(this.contextService.getWorkspace().resource.toString(), this.resource.toString());
```

Comment:
Can you only do this check if the storageService check succeeds, to avoid extra work, since most of the time, it won't be needed?